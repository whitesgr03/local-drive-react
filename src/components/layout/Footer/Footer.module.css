.footer {
	display: flex;
	row-gap: 5px;
	flex: 1 0 auto;
	flex-direction: column;
	align-items: center;
	border-top: 1px solid var(--light-scheme-color);
	padding: 20px;
}

.address-list {
	display: flex;
	column-gap: 30px;
	align-items: center;
}

.address-link {
	display: flex;
	justify-content: center;
	align-items: center;
}

.email {
	--size: var(--icon-size-desktop);

	mask-image: url("data:image/svg+xml,%3Csvg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 36 36'%3E%3Cpath fill='%23000' d='M32 6H4a2 2 0 0 0-2 2v20a2 2 0 0 0 2 2h28a2 2 0 0 0 2-2V8a2 2 0 0 0-2-2m-1.54 22H5.66l7-7.24l-1.44-1.39L4 26.84V9.52l12.43 12.37a2 2 0 0 0 2.82 0L32 9.21v17.5l-7.36-7.36l-1.41 1.41ZM5.31 8h25.07L17.84 20.47Z' class='clr-i-outline clr-i-outline-path-1'/%3E%3Cpath fill='none' d='M0 0h36v36H0z'/%3E%3C/svg%3E");
	width: var(--size);
	height: var(--size);
}

.github {
	--size: var(--icon-size-desktop);

	mask-image: url("data:image/svg+xml,%3Csvg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 24 24'%3E%3Cpath fill='%23000' d='M12 2A10 10 0 0 0 2 12c0 4.42 2.87 8.17 6.84 9.5c.5.08.66-.23.66-.5v-1.69c-2.77.6-3.36-1.34-3.36-1.34c-.46-1.16-1.11-1.47-1.11-1.47c-.91-.62.07-.6.07-.6c1 .07 1.53 1.03 1.53 1.03c.87 1.52 2.34 1.07 2.91.83c.09-.65.35-1.09.63-1.34c-2.22-.25-4.55-1.11-4.55-4.92c0-1.11.38-2 1.03-2.71c-.1-.25-.45-1.29.1-2.64c0 0 .84-.27 2.75 1.02c.79-.22 1.65-.33 2.5-.33s1.71.11 2.5.33c1.91-1.29 2.75-1.02 2.75-1.02c.55 1.35.2 2.39.1 2.64c.65.71 1.03 1.6 1.03 2.71c0 3.82-2.34 4.66-4.57 4.91c.36.31.69.92.69 1.85V21c0 .27.16.59.67.5C19.14 20.16 22 16.42 22 12A10 10 0 0 0 12 2'/%3E%3C/svg%3E");
	width: var(--size);
	height: var(--size);
}

/* Pseudo classes */

.footer-link:hover {
	color: var(--light-scheme-highlight);
}

/* Dark scheme classes */

:global(.dark) .footer {
	border: 0;
}

:global(.dark) .footer-link:hover {
	color: var(--dark-scheme-highlight);
}

@media (width >= 440px) {
	.footer {
		row-gap: 10px;
		padding: 40px 10%;
	}
}

@media (width >= 600px) {
	.email {
		--size: var(--icon-size-mobile);
	}

	.github {
		--size: var(--icon-size-mobile);
	}
}

@media (width >= 700px) {
	.footer {
		background-color: var(--light-scheme-main-bgc);
		padding: 20px;
	}

	:global(.dark) .footer {
		border-top: 1px solid var(--dark-scheme-color);
		background-color: var(--dark-scheme-layout-bgc);
	}

	.email {
		--size: var(--icon-size-desktop);
	}

	.github {
		--size: var(--icon-size-desktop);
	}
}

@media (width >= 800px) {
	.address-list {
		column-gap: 50px;
	}
}
