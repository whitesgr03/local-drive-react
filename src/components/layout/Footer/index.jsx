// Styles
import styles from './Footer.module.css';
import { icon } from '../../utils/icon.module.css';

export const Footer = () => {
	return (
		<footer className={styles.footer}>
			<address>
				<ul className={styles['address-list']}>
					<li>
						<a
							className={`${styles['address-link']} ${styles['footer-link']}`}
							href="mailto:whitesgr03@gmail.com"
						>
							<span className={`${icon} ${styles.email}`} />
						</a>
					</li>
					<li>
						<a
							className={`${styles['address-link']} ${styles['footer-link']}`}
							href="https://github.com/whitesgr03/local-drive-react"
							alt="Local Drive Github Repository"
							rel="noreferrer"
							target="_blank"
							aria-label="Visit Project Repository"
						>
							<span className={`${icon} ${styles.github}`} />
						</a>
					</li>
				</ul>
			</address>
			<p>Â© 2024 Designed & coded by Weiss Bai</p>
		</footer>
	);
};
