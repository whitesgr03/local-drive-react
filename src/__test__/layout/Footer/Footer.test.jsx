import { expect, describe, it } from 'vitest';
import { render } from '@testing-library/react';

import { Footer } from '../../../components/layout/Footer/Footer';

describe('Footer component', () => {
	it('should match snapshot', () => {
		const { asFragment } = render(<Footer />);

		const actual = asFragment();

		expect(actual).toMatchSnapshot();
	});
});
