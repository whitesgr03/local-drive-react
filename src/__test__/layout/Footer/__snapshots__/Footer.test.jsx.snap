// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Footer component > should match snapshot 1`] = `
<DocumentFragment>
  <footer
    class="_footer_bef41e"
  >
    <address>
      <ul
        class="_address-list_bef41e"
      >
        <li>
          <a
            class="_address-link_bef41e _footer-link_bef41e"
            href="mailto:whitesgr03@gmail.com"
          >
            <span
              class="undefined _email_bef41e"
            />
          </a>
        </li>
        <li>
          <a
            alt="Rokulezrive Github Repository"
            aria-label="Visit Project Repository"
            class="_address-link_bef41e _footer-link_bef41e"
            href="https://github.com/whitesgr03/local-drive-react"
            rel="noreferrer"
            target="_blank"
          >
            <span
              class="undefined _github_bef41e"
            />
          </a>
        </li>
      </ul>
    </address>
    <p>
      Â© 2024 Designed & coded by Weiss Bai
    </p>
  </footer>
</DocumentFragment>
`;
